app/controllers/submissions_controller.rb

```diff
@@ -509,39 +507,42 @@ def set_testrun
     }
   end
 
-  def transmit_lti_score
-    responses = send_scores(@submission)
-    failed_users = []
-    too_late = false
-    status = []
-    score_sent = nil
-
-    responses.each do |response|
-      if Lti::ERROR_STATUS.include? response[:status]
-        failed_users << response[:user]
-      elsif response[:score_sent] != @submission.normalized_score # the score was sent successfully, but received too late
-        too_late = true
-        score_sent = response[:score_sent]
-      end
+  def check_scoring_too_late(submit_info, client_socket)
```

Comment:
I am not quite sure how I like the method name: "Checking" something would somehow imply we are returning the result of the check to the callee, wouldn't it? Otherwise, it is pretty close to what we do, but rather inform the learner directly. Maybe I have another idea later today...